private/bufpkg/bufimage/bufimage.go

```diff
@@ -238,6 +238,9 @@ type Image interface {
 	// The path is expected to be normalized and validated.
 	// Note that all values of GetDependency() can be used here.
 	GetFile(path string) ImageFile
+	// Resolver returns a resolver backed by this image.
+	Resolver() protoencoding.Resolver
```

Comment:
I'm really not a fan of this being added as an accessor on `Image` - an `Image` is merely meant to be the wrapper for what amounts to `[]FileDescriptorProto`, in the form of this codebase this feels like something you wouldn't actually include on an `Image`. Is there any way to just return this from `BuildImage`, or does that not accomplish the goal?

Perhaps this is required, I don't know.

Reply:
Talked offline, this is probably the best of the bad options unfortunately